tigerpath/majors_and_certificates/majors/FIT_2018.json

```diff
@@ -0,0 +1,178 @@
+  "urls": [
+    "https://ua.princeton.edu/academic-units/department-french-and-italian",
+    "https://fit.princeton.edu/undergraduate/concentrating/plan-study"
+  ],
+  "contacts": [
+    {
+      "type": "Chair",
+      "name": "Thomas A. Trezise",
+      "email": "ttrezise@princeton.edu"
+    },
+    {
+      "type": "Director of Undergraduate Studies",
+      "name": "Christy Wampole",
+      "email": "cwampole@princeton.edu"
+    }
+  ],
+  "req_list": [
+    {
+      "name": "Prerequisites",
+      "max_counted": 1,
+      "min_needed": 1,
+      "explanation": "Students who wish to concentrate in the department should possess a sound knowledge of either French or Italian by the end of their sophomore year. This can be achieved by completing the language sequence (FRE 107 or 108; ITA 107 or 108), or by demonstrating equivalent qualification. Students are also strongly encouraged to complete at least one advanced-language course (FRE 207, 207F or 208; ITA 207I, 208) prior to admission, or shortly thereafter, which can then be counted toward the concentration. FIT recommends that students complete at least one more 200-level course (such as FRE 211, 215, 217, 221, 222, 224, 225, 230, or ITA 208, 209, 220, 225) before enrolling in a 300-level course.",
+      "completed_by_semester": 4,
+      "req_list": [
+        {
+          "name": "Standard",
+          "max_counted": 1,
+          "min_needed": 1,
+          "explanation": "Students who wish to concentrate in the department should possess a sound knowledge of either French or Italian by the end of their sophomore year. This can be achieved by completing the language sequence (FRE 107 or 108; ITA 107 or 108), or by demonstrating equivalent qualification.",
+          "course_list": [
+            "FRE 107",
+            "FRE 108",
+            "ITA 107",
+            "ITA 108"
+          ]
+        },
+        {
+          "name": "Advanced",
+          "max_counted": 0,
+          "min_needed": null,
+          "explanation": "Students are also strongly encouraged to complete at least one advanced-language course (FRE 207, 207F or 208; ITA 207I, 208) prior to admission, or shortly thereafter, which can then be counted toward the concentration. FIT recommends that students complete at least one more 200-level course (such as FRE 211, 215, 217, 221, 222, 224, 225, 230, or ITA 208, 209, 220, 225) before enrolling in a 300-level course.",
+          "double_counting_allowed": true,
+          "course_list": [
+            "FRE 207",
+            "FRE 207F",
+            "FRE 208",
+            "FRE 211",
+            "FRE 215",
+            "FRE 217",
+            "FRE 221",
+            "FRE 222",
+            "FRE 224",
+            "ITA 207I",
+            "ITA 208",
+            "ITA 209",
+            "ITA 220",
+            "ITA 225"
+          ]
+        }
+      ]
+    },
+    {
+      "name": "Concentration",

```

Comment:
Also, I would put this requirement under the tracks, since the specifics depend on each track, some of which can be verified by TigerPath.

For example, Track 2, "Concentration in two languages, literatures, and cultures", asks for 5 courses in one language and 3 in the other.
So you could have two versions of this track, one with a French emphasis and 3 additional Italian courses, and the other with an Italian emphasis and 3 additional French courses.

For Track 1 you could also have a French version and an Italian version. With 5 departmentals (as above) and 3 cognates (which can't be checked).

Reply:
Maybe it would actually make the most sense to structure the whole major with two large subtrees, one for French and the other for Italian, and then have everything else under one of those two.

Looking at the department website, it looks like that's how the students are categorized, more so than by track. Each concentrator is identified as either a French or an Italian concentrator. It would also simplify the logic and make it more understandable as two possible concentrations.

So the tree would look something like this

FIT `min_needed=1` <-- _specify this explicitly at the top level to override the default of_ `"ALL"`
├─ French `min_needed="ALL"` `max_counted=1`
│   ├─ Prerequisites (only the French courses) `max_counted=1` (see my other comment above re. prereqs)
│   ├─ Departmentals `min_needed=5` `max_counted=1` (as in my comment above but only French classes)
│   └─ Tracks `min_needed=1` `max_counted=1`
│   .  . . ├─ Track 1 `min_needed=3` `max_counted=1` (3 more FIT classes, w/ note that other classes count too)
│   .  . . ├─ Track 2 `min_needed=3` `max_counted=1` (3 Italian classes)
│   .  . . ├─ Track 3 `min_needed=3` `max_counted=1` (3 other classes, cannot be verified)
│   .  . . └─ Track 4 `min_needed=3` `max_counted=1` (3 other art classes, cannot be verified)
├─ Italian `min_needed="ALL"` `max_counted=1`
│   ├─ Prerequisites (only the Italian courses) `max_counted=1`
│   ├─ Departmentals `min_needed=5` `max_counted=1` (as in my comment above but only Italian classes)
│   └─ Tracks `min_needed=1` `max_counted=1`
│   .  . . ├─ Track 1 `min_needed=3` `max_counted=1` (3 more FIT classes, w/ note that other classes count too)
│   .  . . ├─ Track 2 `min_needed=3` `max_counted=1` (3 French classes)
│   .  . . ├─ Track 3 `min_needed=3` `max_counted=1` (3 other classes, cannot be verified)
│   .  . . └─ Track 4 `min_needed=3` `max_counted=1` (3 other art classes, cannot be verified)
├─ Junior Independent Work
└─ Senior Thesis

What do you think?